src/chapter_1.md

```diff
@@ -1 +1,409 @@
+        ConfigTable,
+    },
+    resources::{Resource, ResourceConsumer},
+    units::{PrefixedUnit, Unit, UnitPrefix},
+};
+use serde::{Deserialize, Serialize};
+use std::{fs::File, io::Read, time::Duration};
+
+pub struct ThePlugin {
+    config: Config,
+    metrics: Option<Metrics>,
+}
+
+pub struct Metrics {
+    a_metric: TypedMetricId<u64>,
+}
+#[derive(Serialize, Deserialize, Debug)]
+struct Config {
+    #[serde(with = "humantime_serde")]
+    poll_interval: Duration,
+}
+
+impl Default for Config {
+    fn default() -> Self {
+        Self {
+            poll_interval: Duration::from_secs(1),
+        }
+    }
+}
+
+#[derive(Debug)]
+struct ThePluginSource {
+    random_byte: TypedMetricId<u64>,
+}
+
+impl AlumetPlugin for ThePlugin {
+    fn name() -> &'static str {
+        "ThePlugin"
+    }
+
+    fn version() -> &'static str {
+        env!("CARGO_PKG_VERSION")
+    }
+
+    fn default_config() -> anyhow::Result<Option<ConfigTable>> {
+        Ok(Some(serialize_config(Config::default())?))
+    }
+
+    fn init(config: ConfigTable) -> anyhow::Result<Box<Self>> {
+        let config = deserialize_config(config)?;
+        Ok(Box::new(ThePlugin { config, metrics: None }))
+    }
+
+    fn start(&mut self, alumet: &mut alumet::plugin::AlumetStart) -> anyhow::Result<()> {
+        let my_byte_unit: PrefixedUnit = PrefixedUnit {
+            base_unit: Unit::Byte,
+            prefix: UnitPrefix::Plain,
+        };
+
+        let byte_metric = alumet.create_metric::<u64>("random_byte", my_byte_unit, "Byte randomly get")?;
+        self.metrics = Some(Metrics { a_metric: byte_metric });
+
+        let initial_source = Box::new(ThePluginSource {
+            random_byte: (self.metrics.as_ref().expect("Can't read byte_metric")).a_metric,
+        });
+
+        alumet.add_source(initial_source, TriggerSpec::at_interval(self.config.poll_interval));
+        Ok(())
+    }
+
+    fn stop(&mut self) -> anyhow::Result<()> {
+        Ok(())
+    }
+}
+
+impl Source for ThePluginSource {
+    fn poll(&mut self, measurements: &mut MeasurementAccumulator, timestamp: Timestamp) -> Result<(), PollError> {
+        let mut rng = File::open("/dev/urandom")?; // Open the "/dev/urandom" file to obtain random data
+
+        let mut buffer = [0u8; 8]; // Create a mutable buffer of type [u8; 8] (an array of 8 unsigned 8-bit integer)
+        rng.read_exact(&mut buffer)?; // Read enough byte from the file and store the value in the buffer
+        let value = u64::from_le_bytes(buffer);
+
+        // Print the value of the first byte in the buffer
+        println!("Random u8: {}", buffer[0]);
+        let my_meas_pt = MeasurementPoint::new(
+            timestamp,
+            self.random_byte,
+            Resource::LocalMachine,
+            ResourceConsumer::LocalMachine,
+            value,
+        )
+        .with_attr("double", value.div_euclid(2));
+        measurements.push(my_meas_pt);
+
+        Ok(())
+    }
```

Comment:
You can give this final code at the end of the implementation of poll

Reply:
https://github.com/alumet-dev/developer-book/pull/3#discussion_r1696819867

Reply:
That's not an answer

Reply:
Done, write two times now

Reply:
Yes, and I answered twice